staff/alex-barba/horroy/scripts/horroy.js

```diff
@@ -0,0 +1,739 @@
+    }    
+};
+
+/**
+ * Abstraction of .forEach
+ * 
+ * Functions as .forEach for an array but for a horroy.
+ * 
+ * @param {Function} - To execute on each value of the the horroy.
+ * 
+ * @throws {TypeError} - When callback is not a Function.
+ * 
+ * @returns {*} - Undefined.
+ */
+
+Horroy.prototype.forEach = function(callback) {
+    if (typeof callback !== 'function') throw new TypeError(callback + ' is not a function');
+
+    for (var i = 0; i < this.length; i++) callback(this[i]);
+};
+
+/**
+ * Abstraction of .indexOf
+ * 
+ * Functions as .indexOf for an array but for a horroy.
+ * 
+ * @param {*} - element. Element to locate in the horroy.
+ * @param {Number} - start. Optional. Index to start the search at.
+ * 
+ * @throws {TypeError} - When start is not a Number.
+ * 
+ * @returns {*} - The first index of the element in the horroy; -1 if not found.
+ */
+
+Horroy.prototype.indexOf = function(element, start) {
+    if (!(typeof start === 'number' || start !== 'undefined')) throw new TypeError(start + ' is not a number.')
+
+    start = start? start : 0;
+
+    for(var i=start; i<this.length; i++) {
+        if (this[i] === element){
+            return i;
+        }
+    }
+    return -1;
+};
+
+/**
+ * Abstraction of .join
+ * 
+ * Functions as .join for an array but for a horroy.
+ * 
+ * @param {*} - Optional. Specifies a string to separate each pair of adjacent elements of the array. The separator is converted to a string if necessary.
+ * 
+ * @returns {*} - A string with all horroy elements joined.
+ */
+
+Horroy.prototype.join = function(separator) {
+    var result = '';
+
+    for (let i=0; i<this.length; i++) {
+        if(this[i] === null || this[i] === undefined) this[i] = '';
+        separator = separator? separator : '';
+
+        result += this[i];
+        
+        if (i == this.length-1) {
+            return result;
+        }
+        result += separator;
+    }
+    return result;
+};
+
+/**
+ * Abstraction of .map
+ * 
+ * Functions as .map for an array but for a horroy.
+ * 
+ * @param {Function} - Function that produces an element of the new horroy.
+ * 
+ * @throws {TypeError} - When callback is not a Function.
+ * 
+ * @returns {Horroy} - A new horroy with each element being the result of the callback function.
+ */
+
+Horroy.prototype.map = function(callback) {
+    if (!(callback instanceof Function)) throw TypeError (callback + ' is not a function');
+
+    var res = new Horroy();
+
+    for (var i = 0; i < this.length; i++) {
+        res[i] = callback(this[i]) 
+        res.length++;
+    }
+    return res;
+};
+
+/**
+ * Abstraction of .pop
+ * 
+ * Functions as .pop for an array but for a horroy.
+ * 
+ * @returns {Horroy} - The removed element from the horroy; undefined if the horroy is empty.
+ */
+
+Horroy.prototype.pop = function(){
+    if (this.length === 0) return undefined
+
+    var res = this[this.length-1];
+
+    delete this[this.length-1];
+
+    this.length = this.length--;
```

Comment:
You can shorten this line by just writing: 
```javascript
this.length--;
```

Reply:
You are 100% right... Don't know yet why I didn't do it before! Thanks!